subscription/internal/models/email.go

```diff
@@ -12,8 +13,13 @@ const (
 )
 
 type Email struct {
-	Email     string       `gorm:"primaryKey" json:"email"`
-	Status    Subscription `gorm:"default:subscribed" json:"status"`
-	CreatedAt time.Time    `gorm:"autoCreateTime" json:"createdAt"`
-	DeletedAt *time.Time   `gorm:"index" json:"deletedAt"`
+	Email     string         `gorm:"primaryKey" json:"email"`
+	Status    Subscription   `gorm:"default:subscribed" json:"status"`
+	CreatedAt time.Time      `gorm:"autoCreateTime" json:"createdAt"`
+	DeletedAt gorm.DeletedAt `gorm:"index" json:"deletedAt"`
+}
+
+func (e *Email) BeforeDelete(tx *gorm.DB) (err error) {
+	e.Status = Unsubscribed
```

Comment:
I thought more of a consistency of data, I guess there might be situations in future where subscriptions can be in unsubscribed status but not deleted, on the other hand if the subscription is soft-deleted it most definitely should be in unsubscribed status, possible bottleneck is unscoped selects of subscriptions, it will return both active and soft-deleted rows, to operate correctly with unscoped queries data should be consistent, I guess. But, there is a but) This middleware doesn't really work, because gorm soft deleted actually queries update instead of delete under the hood, so this BeforeDelete function doesn't get triggered, I was planning on figure this out in the future MRs.